<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <arg name="plot_cost" default="false"/>
    <arg name="description_file" default="$(find mppi_pole_cart)/resources/pendulum_cart.urdf"/>
    <!-- TF prefix -->
    <arg name="tf_prefix" default=""/>
    <!-- Load robot description -->
    <param name="robot_description" command="$(find xacro)/xacro $(arg description_file)"/>

    <arg name="learner_output_path" default=""/>
    <arg name="torchscript_model_path" default=""/>
    <arg name="x" default=""/>
    <arg name="x_d" default=""/>
    <arg name="theta" default=""/>
    <arg name="theta_d" default=""/>

    <!--  Run the test node-->
    <node name="pole_cart_data_collection" type="pole_cart_data_collection"
          pkg="mppi_pole_cart" output="screen" required="true">
        <rosparam command="load" file="$(find mppi_pole_cart)/config/params.yaml"/>
        <param name="config_file" value="$(find mppi_pole_cart)/config/params.yaml"/>
        <param name="learned_expert_output_path" value="$(arg learner_output_path)"
            unless="$(eval arg('learner_output_path') == '')" />
        <param name="torchscript_model_path" value="$(arg torchscript_model_path)"
            unless="$(eval arg('torchscript_model_path') == '')"/>
        <param name="initial_position/x" value="$(arg x)"
            unless="$(eval arg('x') == '')"/>
        <param name="initial_position/x_dot" value="$(arg x_d)"
            unless="$(eval arg('x_d') == '')"/>
        <param name="initial_position/theta" value="$(arg theta)"
            unless="$(eval arg('theta') == '')"/>
        <param name="initial_position/theta_dot" value="$(arg theta_d)"
            unless="$(eval arg('theta_d') == '')"/>
    </node>

    <!-- Publish robot state (joints) as TF -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen">
        <param name="publish_frequency" value="100"/>
        <param name="use_tf_static" value="true"/>
        <param name="tf_prefix" value="$(arg tf_prefix)"/>
        <remap from="robot_description" to="robot_description"/>
        <remap from="joint_states" to="/joint_states"/>
    </node>

    <!-- Open Rviz -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find mppi_pole_cart)/config/config.rviz" output="screen"/>

    <group if="$(arg plot_cost)">
        <arg name="mp_config" default="$(find mppi_pole_cart)/resources/rqt_multiplot.xml"/>
        <node name="rqt_multiplot" pkg="rqt_multiplot" type="rqt_multiplot" args="--multiplot-config $(arg mp_config)"
              output="screen"/>
    </group>
</launch>
